File: tests/testdata/codegen/retain-on-delete-pp/dotnet/retain-on-delete.cs
using System.Collections.Generic;
using System.Linq;
using Codeinfra;
using Random = Codeinfra.Random;

return await Deployment.RunAsync(() => 
{
    var foo = new Random.RandomPet("foo", new()
    {
    }, new CustomResourceOptions
    {
        RetainOnDelete = true,
    });

});

File: tests/testdata/codegen/retain-on-delete-pp/go/retain-on-delete.go
package main

import (
	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
	"github.com/khulnasoft/codeinfra/sdk/v3/go/codeinfra"
)

func main() {
	codeinfra.Run(func(ctx *codeinfra.Context) error {
		_, err := random.NewRandomPet(ctx, "foo", nil, codeinfra.RetainOnDelete(true))
		if err != nil {
			return err
		}
		return nil
	})
}
